### PR TITLE
DEV-14726 DEV-14765 Upgrade and pin terraform-aws-modules/key-pair/aws

### PR DESCRIPTION
**Pull request by daniosim**
https://datatamr.atlassian.net/browse/DEV-14765

### CODE REVIEW

**Comment by tamr-teamcity** [comment]
This PR doesn't appear to be linked to a DevOps/SRE jira ticket

**Comment by tamr-teamcity** [comment]
This PR doesn't appear to be linked to a DevOps/SRE jira ticket

**Comment by tamr-teamcity** [comment]
This PR doesn't appear to be linked to a DevOps/SRE jira ticket

**Review by aditimarkale — APPROVED** [review]

File: examples/ephemeral-spark/tamr-vm.tf [hunk 1]
```diff
@@ -3,7 +3,7 @@ module "tamr-vm" {
```
Comment (souza-dan) [inline examples/ephemeral-spark/tamr-vm.tf:3]: If you wanted to close out https://datatamr.atlassian.net/browse/DEV-14726 while we're here, we could update this vm module to be `1.0.2`

**Review by souza-dan — APPROVED** [review]
LGTM, might be able to knock out two tickets in one PR if we upgrade the VM module too https://datatamr.atlassian.net/browse/DEV-14726

**Comment by tamr-teamcity** [comment]
This PR doesn't appear to be linked to a DevOps/SRE jira ticket

**Review by souza-dan — APPROVED** [review]
LGTM

**Comment by tamr-teamcity** [comment]
This PR doesn't appear to be linked to a DevOps/SRE jira ticket

**Pull request merged**